pkg/propertyprovider/azure/controllers/pod.go

```diff
@@ -86,8 +89,12 @@ func (p *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 	// This behavior is consistent with how the Kubernetes CLI tool reports requested capacity
 	// on a specific node (`kubectl describe node` command).
 	//
-	// Note that the tracker will attempt to track the pod even if it has been marked for deletion.
+	// The tracker will attempt to track the pod even if it has been marked for deletion (when it
+	// is actually gone, the pod will be untracked).
 	if len(pod.Spec.NodeName) > 0 && pod.Status.Phase != corev1.PodSucceeded && pod.Status.Phase != corev1.PodFailed {
+		// The pod watcher has field selectors enabled, which will not see pods that should not
```

Comment:
I don't get why we remove the pod if their phase is PodSuucced/Failed.

Reply:
Hi Ryan! Pods in the `Succeeded` or `Failed` state no longer have their resource requests applied, which is why the watcher attempts to ignore them.